client/site-profiler/hooks/use-domain-param.ts

```diff
@@ -7,19 +7,25 @@ import {
 import validateDomain from 'calypso/site-profiler/utils/validate-domain';
 import isSpecialDomain, { prepareSpecialDomain } from '../utils/is-special-domain';
 
-export default function useDomainParam( value: string = '' ) {
+export default function useDomainParam( value = '' ) {
 	const [ domain, setDomain ] = useState( value );
 	const [ isValid, setIsValid ] = useState< undefined | boolean >();
 	const [ isSpecial, setIsSpecial ] = useState( isSpecialDomain( value ) );
 	const [ category, setCategory ] = useState< SPECIAL_DOMAIN_CATEGORY >();
 	const [ readyForDataFetch, setReadyForDataFetch ] = useState( false );
 
 	useEffect( () => {
-		const preparedDomain = prepareDomain( value, isSpecial );
-
+		const isSpecialDomainValue = isSpecialDomain( value );
+		const validateDomainValue = validateDomain( value );
 		// check if domain is special before preparing domain value
-		setIsSpecial( isSpecialDomain( value ) );
-		setIsValid( validateDomain( preparedDomain ) );
+		setIsSpecial( isSpecialDomainValue );
+		setIsValid( validateDomainValue );
+
+		if ( value && ! isSpecialDomainValue && ! validateDomainValue ) {
```

Comment:
Do we need a `value` check here?
The domain can not be special or valid if the value is an empty string or undefined.

Reply:
We need the value check here, because when we clear out the domain, we need to set the `domain` state as empty. Without the value check, clearing the domain here won't work anymore.

Reply:
Cool! 👍  Thanks for the explanation!